default/stringtables/en.txt

```diff
@@ -14332,23 +14338,45 @@ Spinal Antimatter Cannon
 SR_SPINAL_ANTIMATTER_DESC
 A huge spinal mount mass driver firing heavy projectiles made of antimatter. Attacks ships and planets.
 
+SR_GRAV_PULSE
+Graviton Pulsar
+
+SR_GRAV_PULSE_DESC
+'''Graviton pulse causing damage to hull integrity to every target caught in the blast. While still deadly to fighters if the blast is spread, it causes only minor damage to ships or planet defenses.
+
+Graviton pulse inflicts [[encyclopedia DAMAGE_TITLE]] only once in combat, in combat round [[SHORT_RANGE_COMBAT_ROUND]].'''
+
 SR_JAWS
 Jaws
 
 SR_JAWS_DESC
-Jaws of a space monster, can damage ships. Huge space monsters are even capable of eating entire fleets.
+'''Jaws of a space monster, can damage ships and planets. Huge space monsters are even capable of eating entire fleets.
+
+Jaws can inflict [[encyclopedia DAMAGE_TITLE]] starting from combat round [[CLOSE_RANGE_COMBAT_ROUND]] as the targets are too far away in earlier rounds.'''
 
 SR_SPINES
 Spines
 
 SR_SPINES_DESC
-Spines of a space monster, a powerful strike of the space monster can deal big damage to a ship.
+'''Spines of a space monster, a powerful strike of the space monster can deal big damage to a ship.
+
+Spines inflict [[encyclopedia DAMAGE_TITLE]] starting from combat round [[CLOSE_RANGE_COMBAT_ROUND]] as the targets are too far away in earlier rounds.'''
 
 SR_TENTACLE
 Tentacle
 
 SR_TENTACLE_DESC
-Tentacle of a kraken, can damage ships.
+'''Tentacle of a kraken, primarily do damage to ships. Is also able to attack fighters and planets.
+
+Tentacles can inflict [[encyclopedia DAMAGE_TITLE]] to ships starting from combat round [[CLOSE_RANGE_COMBAT_ROUND]] as the targets are too far away in earlier rounds. Tentacles start to destroy fighters when they come in range in [[SHORT_RANGE_COMBAT_ROUND]]. Tentacles only attack fighters or planets if there are no ships to target.'''
```

Comment:
```suggestion
Tentacles can inflict [[encyclopedia DAMAGE_TITLE]] to ships, fighters, and planets. In the first combat round, they cannot reach anything and shake the tentacles in anger, at the [[SHORT_RANGE_COMBAT_ROUND]] round they can smash fighters. Later on, they can reach any targets but prefer crushing ships first.
```

I'd prefer removing [[SHORT_RANGE_COMBAT_ROUND]] and [[CLOSE_RANGE_COMBAT_ROUND]] from string tables. 
Since we can count rounds from the beginning it is ok to embed them into the translation.

short = 2 and close = 3 could be confusing to translater (as it was for me at the beginning).  Using first, second and third as names is probably a better option.

Reply:
I see your pain. Not completly convinced if THIRD_COMBAT_ROUND is good though. The whole point of the macro is to avoid magic numbers. What if the close range round is changed from 3 to e.g. 5? Then all the translators would need to change all occurences of THIRD_COMBAT_ROUND to FIFTH_COMBAT_ROUND

Reply:
How about something like THIRD_WAVE_COMBAT_ROUND (at the moment 3, but could also become e.g. 5) or stressing that the contents of the macro is a number by using something like SHORT_RANGE_COMBAT_ROUND_NUMBER.
One could also make this clearer by shifting this from a macro to a value ref (or a rule) using a `[value SHORT_RANGE_COMBAT_ROUND]` expression in the stringtables.

Reply:
> The whole point of the macro is to avoid magic numbers. What if the close range round is changed from 3 to e.g. 5? 

It is hardcoded in the script: `(CombatBout > 1)`   Also that means that close range for tentacles is not the same close range for Jaws.    

Terms *RANGE_COMBAT and *WAVE_COMBAT_ROUND are not a part of focs-script language (at least of parts we are talking about).  So adding new terms with new meanings may be confusing to the translations.   Actually, I don't know how to communicate this information to them. This example is not enough to share an idea behind waves: `THIRD_WAVE_COMBAT_ROUND = 5`.

I totally agree that a new layer of abstractions could make an update of the translation much easier, but we cannot guarantee that it will be maintained properly.  And keep in mind having them in a translation and not having them in focs-scripts make things even more complicated than using combat rounds in the text.

> using a [value SHORT_RANGE_COMBAT_ROUND]

Can we use this ref in focs-scripts?

Reply:
> > The whole point of the macro is to avoid magic numbers. What if the close range round is changed from 3 to e.g. 5?
> 
> It is hardcoded in the script: `(CombatBout > 1)` Also that means that close range for tentacles is not the same close range for Jaws.

well. as i said fighter fluff sais those are really close/able to penetrate shields starting in bout 2 (while all other weapons are not able to do that). so those two weapons are consistent (jaws are not targeting fighters at all, while tentacles and  only target fighters in bout 2). Spines are like jaws. SR_THRASHING_BODY is not consistent as it also targets fighters, but only starting from bout 2. I could fluff that away or make it the same complex attack pattern like tentacles.

> 
> Terms *RANGE_COMBAT and *WAVE_COMBAT_ROUND are not a part of focs-script language (at least of parts we are talking about). So adding new terms with new meanings may be confusing to the translations. Actually, I don't know how to communicate this information to them. This example is not enough to share an idea behind waves: `THIRD_WAVE_COMBAT_ROUND = 5`.

The right place to talk about complex topics would be in an encyclopedia entry. The closest we have currently is default/scripting/encyclopedia/game_concepts/ship_management/DAMAGE.focs.txt i guess, so the DAMAGE_TITLE and DAMAGE_TEXT, but we could write another entry about combat/targeting/ranges/waves.

Note that FOCS also has no notion of colony/outpost. One is a planet with some population, the other without. So I expect more knowledge from somebody who is coding FOCS.

> 
> And keep in mind having them in a translation and not having them in focs-scripts make things even more complicated than using combat rounds in the text.

wasnt sure if i should introduce that into FOCS; could be done by a macro, a rule or a named value ref. Or some combination of rule and macro/named value ref.
The almost-zero-overhead version is the macro variant.

@geoffthemedio is it ok for you if i introduce a rule for defining the first combat round in which the targets come into range? like RULE_FIRST_COMBAT_ROUND_IN_CLOSE_TARGETING_RANGE and RULE_FIRST_COMBAT_ROUND_IN_SHORT_TARGETING_RANGE. The upside of having this as game rules is that you can adjust it if you adjust the RULE_NUM_COMBAT_ROUNDS.

Adjusting the number of combat rounds strongly shifts the damage output (three bouts is a strong nerf for unhidden close range weapons, while making hidden ones "invincible"; more than four bouts means close range weapons get a boost). The current implementation follows the idea that we mainly stick with four bouts as was discussed in the forums. If we have the game rules one can set e.g. 8 combat rounds, short targeting start in bout 3 and close targeting in bout 5 and the overall damage would be restored. If you want to compensate first-strike advantage you could make it 1/2-3/4-7 instead of 1-2/3-4/5-8.

Something like RULE_COMBAT_ROUND_WHEN_SECOND_ATTACK_WAVE_STARTS would also be possible but I think the wave metaphor is worse than the targeting range one.

> 
> > using a [value SHORT_RANGE_COMBAT_ROUND]
> 
> Can we use this ref in focs-scripts?

This can only be defined in the focs-scripts; it can be read from stringtables.